tests/xdvipdfmx-outputs.rs

```diff
@@ -0,0 +1,149 @@
+// Copyright 2016-2017 the Tectonic Project
+// Licensed under the MIT License.
+extern crate flate2;
+#[macro_use] extern crate lazy_static;
+extern crate tectonic;
+
+use std::ffi::OsStr;
+use std::fs::File;
+use std::io::{Read, Write};
+use std::path::{Path, PathBuf};
+use std::sync::Mutex;
+use std::env;
+use std::collections::HashSet;
+
+use tectonic::config::PersistentConfig;
+use tectonic::engines::NoopIoEventBackend;
+use tectonic::io::{FilesystemIo, GenuineStdoutIo, IoStack, MemoryIo};
+use tectonic::io::testing::SingleInputFileIo;
+use tectonic::status::NoopStatusBackend;
+use tectonic::XdvipdfmxEngine;
+
+const TOP: &'static str = env!("CARGO_MANIFEST_DIR");
+
+lazy_static! {
+    static ref LOCK: Mutex<u8> = Mutex::new(0u8);
+}
+
+fn read_file<P: AsRef<Path>>(path: P) -> Vec<u8> {
+    let mut buffer = Vec::new();
+    let mut f = File::open(&path).unwrap();
+    f.read_to_end(&mut buffer).unwrap();
+    buffer
+}
+
+
+pub fn test_file(name: &OsStr, expected: &Vec<u8>, observed: &Vec<u8>) {
+    if expected == observed {
+        return;
+    }
+
+    // For nontrivial tests, it's really tough to figure out what
+    // changed without being able to do diffs, etc. So, write out the
+    // buffers.
+
+    {
+        let mut n = name.to_owned();
+        n.push(".expected");
+        let mut f = File::create(&n).expect(&format!("failed to create {} for test failure diagnosis", n.to_string_lossy()));
+        f.write_all(expected).expect(&format!("failed to write {} for test failure diagnosis", n.to_string_lossy()));
+    }
+    {
+        let mut n = name.to_owned();
+        n.push(".observed");
+        let mut f = File::create(&n).expect(&format!("failed to create {} for test failure diagnosis", n.to_string_lossy()));
+        f.write_all(observed).expect(&format!("failed to write {} for test failure diagnosis", n.to_string_lossy()));
+    }
+
+
+    println!("hex'd observed output:");
+    for &byte in observed {
+        print!("{:X}", byte);
+    }
+    println!("\n");
```

Comment:
Because the contents are saved to disk, this bit seems unnecessary do me. It will dump a whole lot to the screen if we start testing big files.